shell/components/fleet/dashboard/ResourceDetails.vue

```diff
@@ -0,0 +1,182 @@
+<script>
+import { FLEET } from '@shell/config/types';
+import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import FleetResources from '@shell/components/fleet/FleetResources';
+import FleetRepo from '@shell/components/formatter/FleetRepo.vue';
+
+export default {
+  name: 'FleetDashboardResourceDetails',
+
+  components: {
+    LabeledSelect,
+    FleetRepo,
+    FleetResources
+  },
+
+  props: {
+    value: {
+      type:     Object,
+      required: true
+    },
+
+    statePanel: {
+      type:     Object,
+      required: true
+    },
+
+    workspace: {
+      type:     Object,
+      required: true
+    }
+  },
+
+  data() {
+    return {
+      FLEET,
+      cluster: null
+    };
+  },
+
+  mounted() {
+    this.cluster = this.clusterOptions[0]?.value;
+  },
+
+  computed: {
+    clusterOptions() {
+      return this.value.targetClusters.map((cluster) => ({
+        label: cluster.nameDisplay,
+        value: cluster.id
+      }));
+    },
+
+    noResources() {
+      return !this.value.resourcesStatuses?.length;
+    },
+  },
+
+  methods: {
+    closePanel() {
+      this.$store.commit('slideInPanel/close');
+    }
+  }
+};
+</script>
+
+<template>
+  <div class="details-panel">
+    <div
+      class="header"
+      :data-testid="'fleet-dashboard-resource-details-header'"
+    >
+      <div class="title">
+        <i :class="value.dashboardIcon" />
+        <router-link
+          class="label"
+          :to="value.detailLocation"
+        >
+          {{ value.id }}
+        </router-link>
+        <i
+          v-if="statePanel.id !== 'success'"
+          class="ml-5 state-icon"
+          :class="statePanel.icon"
+          :style="{ color: statePanel.color }"
+        />
+      </div>
+      <i
```

Comment:
Consider if a button works better than a clickable icon. Using a button would cut down on the amount of accessibility concerns in this area.